### PR TITLE
Add max31855 driver

### PR DESCRIPTION
**Pull request by rusefillc**

### CODE REVIEW

**Comment by rusefillc** [comment]
@dron0gus :)

**Comment by dron0gus** [comment]
Fixed.

File: firmware/max31855.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	float temp[EGT_CHANNELS];
+
+	Max31855Thread(const char* name)
+		: ThreadController(name, MAX31855_THREAD_PRIO)
+	{
+	}
+	void ThreadTask() override;
+};
+
+extern Max31855Thread EgtThread;
```
Comment (mck1117) [inline firmware/max31855.h:35]: please no extern

File: firmware/max31855.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+} Max31855State;
+
+class Max31855Thread : public ThreadController<MAX31855_THREAD_STACK> {
+private:
+	int spi_rx(SPIDriver *spi, int ch, uint32_t *data);
+	int Read(int ch);
+public:
+	Max31855State state;
+	float int_temp[EGT_CHANNELS];
+	float temp[EGT_CHANNELS];
```
Comment (mck1117) [inline firmware/max31855.h:26]: Why do we have multiple arrays of things inside an object, instead of an array of objects each with its own things?

Comment (mck1117) [reply firmware/max31855.h:26]: the 31855 is a single channel device - the driver should be a single channel driver that we make N instances of

Comment (dron0gus) [reply firmware/max31855.h:26]: To have one thread for all instances.

Comment (mck1117) [reply firmware/max31855.h:26]:
```
void thread() {
    while (true) {
        for (int i = 0; i < instCount; i++) {
            egt[i].update();
        }
    }
}
```

Comment (mck1117) [reply firmware/max31855.h:26]: And if that then means the thread class has one function and no state - it should be a plain function instead

File: firmware/max31855.h (hunk 1 above)
Comment (rusefillc) [reply firmware/max31855.h]: fixed

File: firmware/max31855.h (hunk 2 above)
Comment (rusefillc) [reply firmware/max31855.h]: fixed?

Comment (mck1117) [reply firmware/max31855.h]: fixed!

**Pull request merged**